### ISSUE TITLE
Explore using maven for building project?

### ISSUE THREAD

**Issue by ngzhian** [TEAMMATES/teammates#3962]
It started with a question on why we download and extract a zip file when setting up the dev environment. Learned that maven/gradle is usually used for such things, and prompted to file this issue.
Interested in how the current developers feel about this.
I think this will help #3295 move forward as well.

**Comment by damithc** [comment]
Some history, for future reference:
We used Maven for a period in the past, about 6 months. We switched back to providing a zip file because Maven created more problems than it solved. Some of the reasons:
* Because our folder structure did not match the structure expected by maven, it was retrofitted in a hackish way that required copying files to another location: one copy for eclipse and one copy for maven. While the copying was done by maven automatically, having two copies complicated hot fixing. e.g. When a jsp file is modified, the modification did not show up in the dev server unless we shutdown the dev server and run maven again. This slowed down UI tweaks.
* When testing was done using maven, test failures were hard to troubleshoot. With eclipse, we can just double click to reach the offending test and see the expected/actual diff, rerun failed tests easily etc. With maven, it was not as straight forward; we had to dig through lengthy test reports instead.
* At the time the maven plugin for eclipse was buggy. That was one reason why we could not get eclipse and maven to play together nicely. Whenever something went wrong, we had to answer an extra question 'was it something to do with maven?' which required everyone to have a decent knowledge of maven. Unfortunately haven has a steep learning curve.

In the end, maven created more problem than it solved, so we were like 'screw it, we don't need this grief'. Because we update jar files only about twice an year, providing them as a zip file was a simple and good enough solution. Because TEAMMATES does not use CI, having a full-fledged build tool like maven was not a must for us so it was no big loss.

Having said that, I'm open to going back to a build tool if it helps. I don't mind having another look at how maven, gradle, or other similar tool can help TEAMMATES. HubTurbo uses gradle and it seems to work well for that project. We can also consider using a build tool just for managing dependencies, but not using it to manage building, testing, deploying, etc. Eclipse seems good enough for building/testing/deploying.

On a general note (and this is my advice to any new developer), one needs to be quite familiar with the current process/infrastructure, project needs, workflow, legacy code etc before one can propose a process/infrastructure improvement that fits the project well. So I suggest you hold off this for now, until you have worked in the project for a few months first.
